### PR TITLE
[slack] Query bank for Assistance V1

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Initial query bank extraction and surfacing in Slack
- Enabling message blocks 
- queries.json --> Saved queries for sharing with other users

## How was this patch tested?

Screenshot 2021-05-24 at 5 18 46 PM

### CODE REVIEW

File: desktop/core/src/desktop/lib/botserver/data/queries.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "name": "Customer Ticket Analysis",
+    "query": "SELECT 1"
+  },
+  {
+    "name": "Employee Sales",
+    "query": "SELECT 2"
+  },
+  {
+    "name": "Quaterly Revenue Query",
+    "query": "SELECT 3"
```
Comment (romainr) [inline desktop/core/src/desktop/lib/botserver/data/queries.json:12]: Put queries to sample Hue sample tables? (sample_07, weblogs) With `${variables}` too?

Comment (romainr) [reply desktop/core/src/desktop/lib/botserver/data/queries.json:12]: PS: we can probably move this out of into a `assistant` in https://github.com/cloudera/hue/tree/master/desktop/libs/metadata/src/metadata as this is can be re-used within the Editor itself, not just Slack.

File: desktop/core/src/desktop/lib/botserver/views.py [hunk 2]
```diff
@@ -79,14 +80,64 @@ def parse_events(request, event):
...
+  if text and 'query bank' in text:
+    slack_user = check_slack_user_permission(host_domain, user_id)
+    user = get_user(channel_id, slack_user)
+
+    handle_query_bank(channel_id, user_id)
+
+  
+def handle_query_bank(channel_id, user_id):
+  with open(os.path.join(os.path.dirname(__file__), 'data/queries.json')) as file:
+    data = json.load(file)
```
Comment (romainr) [inline desktop/core/src/desktop/lib/botserver/views.py:105]: All the query bank management should be out in his own utils not tangle with slack logic, e.g.

get_query_suggestions(dialect, query_search) --> [] of queries

(in the new `assistant` directory)

File: desktop/core/src/desktop/lib/botserver/views.py [hunk 3]
```diff
@@ -79,14 +80,64 @@ def parse_events(request, event):
...
+        "type": "section",
+        "text": {
+          "type": "mrkdwn",
+          "text": "*Statement:*\n ```{query}```".format(query=query.get('query'))
+        },
+      }
+      message_block.append(statement)
+
+    _send_message(channel_id, block_element=message_block)
+
```
Comment (romainr) [inline desktop/core/src/desktop/lib/botserver/views.py:141]: Where are the tests?

**Review by romainr — APPROVED** [review]
This is a great start!

Note: we should find a way to have links to open them up in the editor (or example in Slack). 
e.g.
`..../create?dialect=hive&statement=SELECT ...`
I thought we had it but could not find it. Could do js or API side `notebook.views.py`
`async newNotebook(connectorId, callback, queryTab, blank) {` or 

In practice those queries are pretty long, so better to use more realistic examples and see how it fits/feel.
Will for sure at least need a dialect metadata. 

Maybe filtering down the road, multi banks, e.g. '@hue query bank sfdc', '@hue queries customer logs' etc

'queries' or something similar than 'query bank' too. (and add the list of options the bot supports in its initial welcome message or when asking him '@hue help'

**Comment by Harshg999** [comment]
- In its own `assistant` directory with its own utils
- Populated `queries.json` with some sample queries for a better look/feel
- Added `dialect` and `desc` fields. However, currently not showing up in Slack
- Improved a better app_mention name (@hue query bank ---> @hue queries)

query-bank

**Comment by Harshg999** [comment]
Will squash and merge after unit tests

File: desktop/core/src/desktop/lib/botserver/views.py (hunk 3 above)
Comment (Harshg999) [reply desktop/core/src/desktop/lib/botserver/views.py:134]: Added unit tests

**Pull request merged**